### PR TITLE
feat(network): automatic Antenna Criteria check

### PR DESCRIPTION
**Pull request by LeonVreling**

### CODE REVIEW

**Comment by LeonVreling** [comment]
### List of used permissions:

**_NetCom:_**
- global:manage_network:antenna_criteria
- global:manage_network:communication

**_ND:_**
- _All of NetCom_
- global:manage_network:communication_exception
- global:manage_network:board_election
- global:manage_network:members_list
- global:manage_network:events
- global:manage_network:agora_attendance
- global:manage_network:development_plan
- global:manage_network:fulfilment_report

**_FD:_**
- global:manage_network:antenna_criteria
- global:manage_network:membership_fee


EDIT: Update according to commit [#ecac9ab](https://github.com/AEGEE/frontend/pull/2036/commits/ecac9ab562952013996eace0ff3f333eff6b3ba0)

**Comment by LeonVreling** [comment]
@WikiRik I just saw there are still two TODO's in the code for the checking of the `Events` criterion, but I think it would still be good to have a review since the list of changes is quite long 😁

**Comment by WikiRik** [comment]
Yes, I think a review is good to do. I should be able to work on that this weekend

**Comment by WikiRik** [comment]
@LeonVreling can you make a PR in core for the new permissions (and the other permissions added in network earlier) and NetCom/ND/FD users? https://github.com/AEGEE/core/blob/stable/scripts/seed.js

File: src/router/routes.js [hunk 1]
```diff
@@ -841,6 +841,14 @@ module.exports = [
       label: 'View current boards'
     }
   },
+  {
+    name: 'oms.network.ac_check',
+    path: '/antenna_criteria_check',
```
Comment (WikiRik) [inline src/router/routes.js]:
```suggestion
    path: '/network/ac_check',
```

File: src/views/network/AntennaCriteriaCheck.vue [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+    toggleShowDetails () {
+      this.showDetails = !this.showDetails
+    },
+    toggleHideSafeLocals () {
+      this.hideSafeLocals = !this.hideSafeLocals
+    },
+    fetchAgorae () {
+      this.isLoading = true
+      this.axios.get(this.services['statutory'], { params: { type: 'agora' } }).then((response) => {
+        this.agorae = response.data.data
```
Comment (WikiRik) [inline src/views/network/AntennaCriteriaCheck.vue:218]: Benefit of testing it on a relatively empty database is seeing how it works without any Agorae. Can you make it throw an error here if `this.agorae` is empty? Now it shows an error for 'Could not fetch bodies'

**Review by WikiRik — COMMENTED** [review]
I'm going to take some time to review the backend PRs first and then set up more data to test with before I review this further

---

This is what I see if I only have the NetCom permissions created (and all that is in the core seeder), is that correct?
image

**Comment by LeonVreling** [comment]
> @LeonVreling can you make a PR in core for the new permissions (and the other permissions added in network earlier) and NetCom/ND/FD users? https://github.com/AEGEE/core/blob/stable/scripts/seed.js

The pull request for core can be found at https://github.com/AEGEE/core/pull/972

File: src/views/network/AntennaCriteriaModal.vue [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        const permission = 'set' + criterion.charAt(0).toUpperCase() + criterion.slice(1)
+        if (this.can[permission] && (this.antennaCriteria[criterion] !== this.local.antennaCriteria[criterion] || this.comments[criterion] !== this.local.comments[criterion])) {
+          promises.push(this.setAntennaCriterionFulfilment(criterion))
+        }
+      }
+
+      await Promise.all(promises).then(() => {
+        this.isLoading = false
+        this.showSuccess('Antenna Criteria fulfilment updated.')
+        // this.router.go(0)
```
Comment (WikiRik) [inline src/views/network/AntennaCriteriaModal.vue]:
```suggestion
```

File: src/views/core/bodies/ViewBoards.vue [hunk 4]
```diff
@@ -109,7 +128,7 @@ export default {
     },
     fetchData () {
       // Get boards of the body
-      this.axios.get(this.services['network'] + '/bodies/' + this.$route.params.id + '?sort=start_date&direction=desc').then((response) => {
+      this.axios.get(this.services['network'] + '/bodies/' + this.$route.params.id + '/boards?sort=start_date&direction=desc').then((response) => {
```
Comment (WikiRik) [inline src/views/core/bodies/ViewBoards.vue:131]: I think I need to merge https://github.com/AEGEE/network/pull/599 before this works. Maybe move the changes to this file to another PR so we can merge the actual AC checks first

Comment (WikiRik) [reply src/views/core/bodies/ViewBoards.vue:131]: Oh wait, it needs that PR either way to properly display the AC page. We can keep this

Comment (LeonVreling) [reply src/views/core/bodies/ViewBoards.vue:131]: Ah I do agree that it would have been cleaner to make this a separate PR

**Review by WikiRik — APPROVED** [review]
We shouldn't deploy this until https://github.com/AEGEE/network/pull/599 is merged, but this looks good

**Pull request merged**